src/blockdata/transaction.rs

```diff
@@ -609,6 +616,60 @@ impl Transaction {
     pub fn is_explicitly_rbf(&self) -> bool {
         self.input.iter().any(|input| input.sequence < (0xffffffff - 1))
     }
+
+    /// Returns the nLockTime as an absolute timelock.
+    pub fn absolute_lock_time(&self) -> timelock::Abs {
+        timelock::Abs::from(self.lock_time)
+    }
+
+    /// Returns `true` if this transactions nLockTime should be ignored.
+    pub fn ignore_lock_time(&self) -> bool {
+        if self.lock_time == 0 {
+            return true;
+        }
+        self.input.iter().all(|input| input.sequence == 0xffffffff)
+    }
+
+    /// Returns the most strict relative timelocks that must be satisfied for this transaction.
+    ///
+    /// # Returns
+    ///
+    /// `(t1, t2)`, where t1 is the most strict block based relative timelock and t2 is the most
+    /// strict time based relative timelock.
+    ///
+    /// `None` if timelock was not found, `Some(t)` if a timelock was found (including zero value
+    /// timelocks).
+    pub fn maximum_relative_lock_time(&self) -> (Option<timelock::Rel>, Option<timelock::Rel>) {
```

Comment:
This signature is kind of meh from a type-system expression perspective. Especially the fact that we need to document which argument is which. I sometimes wish Rust had anonymous structs, so this could all be:

```
pub fn maximum_relative_lock_time(&self) ->  { block: Option<timelock::Rel>, time: Option<timelock::Rel>}
```

avoiding to have to name this type. But since it can't I would consider

```
struct EffectiveTimelock {
  pub block: Option<timelock::Rel>,
  pub time: Option<timelock::Rel>,
}
```

just to name fields.

Reply:
The alternative would be to have 

```
enum Relative {
  Block(Blocks),
  Time(Seconds),
}

struct Blocks(u32);

struct Seconds(u32);
```

and then this function could return:

```
pub fn maximum_relative_lock_time(&self) -> (Option<Blocks>, Option<Seconds>);
```

but even I have to admit that the multiplication of types one has to juggle becomes rather annoying.

Again, had Rust had "enum variant types" feature https://github.com/rust-lang/lang-team/issues/122 it would possible to:

```
pub fn maximum_relative_lock_time(&self) -> (Option<Relative::Blocks>, Option<Relative::Seconds>);
```

without introducing any new types.